$OpenBSD: patch-jdk_src_share_classes_sun_net_spi_DefaultProxySelector_java,v 1.1 2013/02/01 15:34:35 kurt Exp $

Crash due to missing synchronization on gconf_client.
Upstream CR 7188755.

--- jdk/src/share/classes/sun/net/spi/DefaultProxySelector.java.orig	Fri Oct 26 14:26:02 2012
+++ jdk/src/share/classes/sun/net/spi/DefaultProxySelector.java	Mon Jan 28 20:54:44 2013
@@ -435,6 +435,6 @@ public class DefaultProxySelector extends ProxySelecto
         return false;
     }
 
-    private native static boolean init();
-    private native Proxy getSystemProxy(String protocol, String host);
+    private synchronized native static boolean init();
+    private synchronized native Proxy getSystemProxy(String protocol, String host);
 }
